### ISSUE TITLE
flux_spectrum returns exactly 0.0 with no warning when per-cell E×H* products underflow float32 (subnormals flushed, x64 disabled)

### ISSUE THREAD

**Issue by bk-squared** [bk-squared/rfx#304]
## Summary

With `jax_enable_x64` disabled (the default), `flux_spectrum` returns **exactly 0.0 at every frequency** — not a small number — when the per-cell Poynting products fall below the float32 minimum normal (~1.1755e-38). No warning or guard fires anywhere on this path, so a physically tiny-but-nonzero flux silently becomes a hard zero, which then poisons downstream ratios (two-run reference normalization, dB conversion, transmission fractions).

## Mechanism (verified against source + empirically)

1. `init_flux_monitor` requests `complex128` accumulators (`rfx/probes/probes.py:548`), but with x64 disabled JAX truncates this to `complex64`. The only signal is the generic JAX "Explicitly requested dtype ... will be truncated" `UserWarning` at monitor creation — it says nothing about flux dynamic range and is easily filtered/missed. `dA` is explicit `float32` (`probes.py:553-554`).
2. `flux_spectrum` (`probes.py:615-616`) computes per-cell products in float32 complex:
   ```python
   integrand = mon.e1_dft * jnp.conj(mon.h2_dft) - mon.e2_dft * jnp.conj(mon.h1_dft)
   return jnp.real(jnp.sum(integrand * mon.dA, axis=(-2, -1)))
   ```
3. XLA on CPU flushes subnormal float32 results to zero. Empirical check (JAX CPU, x64 off):
   ```python
   jnp.asarray(1e-20+0j, jnp.complex64) * jnp.conj(jnp.asarray(1e-22+0j, jnp.complex64))
   # -> exactly 0j            (NumPy gives the subnormal 1e-42)
   # 1e-15 * 1e-22 = 1e-37    -> survives (above min normal 1.1755e-38)
   ```
   So whenever every per-cell |E_dft|·|H_dft| product is below ~1.18e-38, the flux sum is exactly 0.0.

Important detail: the **accumulators themselves remain healthy** — each field DFT amplitude (field x dt x window, summed over steps) is typically far above min-normal. It is the *pairwise product* that underflows, so there is no visibly broken intermediate to notice; the failure only appears at the final reduction. We found no underflow/dynamic-range guard on this path (`probes.py`, `_preflight.py`, `_execute.py`).

## Why fine grids hit this readily (amplifier)

A fixed-amplitude single-cell soft source has an effective dipole moment proportional to the cell volume (~dx^3), so fields at a fixed monitor plane scale ~dx^3 and the per-cell E x H* product scales **~dx^6** (with an additional dt^2 factor from the DFT kernel). Halving dx costs ~64x in per-cell integrand, so a convergence study with a fixed source amplitude can cross the flush threshold between two ladder rungs — the coarse run gives a finite flux and the fine run gives exactly 0.0.

## Repro sketch

```python
# x64 disabled (default)
sim = rfx.Simulation(...)                 # fine grid, open boundaries
sim.add_source(..., amplitude=weak)       # single-cell soft source, fixed amplitude
sim.add_flux_monitor(...)                 # accumulators end up complex64
res = sim.run(n_steps)
# flux spectrum: exactly array([0., 0., ...]) — not small, exactly zero
```

## Possible directions (no strong preference)

- A cheap post-hoc guard in `flux_spectrum` or at result assembly: if the returned flux is exactly 0.0 at all frequencies while the accumulators are nonzero, emit a warning naming the float32 dynamic-range limit and suggesting a larger source amplitude (flux scales quadratically, so rescaling is exact) or scoped x64.
- Upcasting the reduction only (cast the four DFT planes to `complex128`/`float64` inside `flux_spectrum` before the product) extends the *range* even though upstream fields stay float32. Note this is distinct from the previously investigated complex128 precision change on the band-edge path — here the problem is range (flush-to-zero), not precision, and the upcast is confined to one plane-sized reduction.
- At minimum, a documentation note on the flux-monitor page about the float32 product dynamic-range floor and the ~dx^6 scaling of single-cell fixed-amplitude sources.

Measured provenance: flush behavior reproduced on this repo's CPU JAX install (products at 1e-40/1e-42 -> exactly 0j; 1e-37 survives), during the NTFF directivity battery session.
_Found during the NTFF directivity validation battery work (PR #302); the battery includes a float64-recompute witness gate as a workaround. Mechanism verified against source + an empirical subnormal-flush repro._
